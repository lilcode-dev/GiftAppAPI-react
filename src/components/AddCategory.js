import React, { useState } from 'react'
import './AddCategory.css'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('') // useState() is undefined! first state of inputvalue

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }
    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories((values) => [inputValue, ...values])
            setInputValue('')
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}

//CHECK LATER!!!!!!!
// inputValue.trim().length > 2
//     ? ( setCategories((values) => [...values, inputValue]),
//       setInputValue(''))
//     : false
