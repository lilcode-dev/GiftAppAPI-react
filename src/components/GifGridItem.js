import React from 'react';
import PropTypes from 'prop-types'


export const GifGridItem = ({ url, title }) => {
    return (
        <div className="card animate__animated animate__backInLeft">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes= {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

