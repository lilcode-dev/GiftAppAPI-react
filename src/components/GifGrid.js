import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/getGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({ category }) => {
    // const [images, setImages] = useState([])
    // useEffect(() => {
    //     getGifs(category).then(setImages);
    //     // other 
    //     // getGifs(category).then((imgs) => setImages(imgs))
        
    // }, [])
    const {data:images, loading} = useFetchGifs(category);
    return (
        <>
            <h3 className='animate__animated animate__jello'>{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="card-grid animate__animated">
                {images.map((img) => (
                    <GifGridItem key={img.id} {...img} />
                ))}
            </div>
        </>
    )
}
