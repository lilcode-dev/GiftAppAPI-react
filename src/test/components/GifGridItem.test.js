import React from 'react'
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('test on <GifGridItem />', () => {
    const title = 'A title'
    const url = 'https://klk.com'
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('should display the component correctly', () => {
        expect(wrapper).toMatchSnapshot()
    });
    test('should have a paragraph with the title', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title); 
        
    });
})
