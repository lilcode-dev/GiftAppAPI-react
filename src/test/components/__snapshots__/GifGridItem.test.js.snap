// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifGridItem /> should display the component correctly 1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <img
    alt="A title"
    src="https://klk.com"
  />
  <p>
    A title
  </p>
</div>
`;
